File: backend/location-service/src/main/java/com/trucktrack/location/service/DashboardService.java
package com.trucktrack.location.service;

import com.trucktrack.location.dto.*;

import java.util.List;

/**
 * T011: Dashboard service interface for aggregating dashboard data.
 * Feature: 022-dashboard-real-data
 */
public interface DashboardService {

    /**
     * Get KPI metrics for the dashboard.
     * @param userGroups comma-separated list of group UUIDs the user has access to
     * @return KPI data including total trucks, active trucks, trips today, alerts
     */
    DashboardKpiDTO getKpis(String userGroups);

    /**
     * Get fleet status breakdown for donut chart.
     * @param userGroups comma-separated list of group UUIDs the user has access to
     * @return Fleet status with counts and percentages per status
     */
    FleetStatusDTO getFleetStatus(String userGroups);

    /**
     * Get recent activity events.
     * @param userGroups comma-separated list of group UUIDs the user has access to
     * @param limit maximum number of events to return
     * @return List of recent activity events
     */
    List<ActivityEventDTO> getRecentActivity(String userGroups, int limit);

    /**
     * Get performance metrics for specified period.
     * @param userGroups comma-separated list of group UUIDs the user has access to
     * @param period "week" or "month"
     * @return Performance metrics for the period
     */
    PerformanceMetricsDTO getPerformanceMetrics(String userGroups, String period);

    /**
     * Get all dashboard data in a single call.
     * @param userGroups comma-separated list of group UUIDs the user has access to
     * @param userId the current user's ID
     * @param performancePeriod "week" or "month" for performance metrics
     * @return Aggregated dashboard data
     */
    DashboardDataDTO getDashboardData(String userGroups, String userId, String performancePeriod);
}

File: frontend/src/app/services/dashboard.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';
import {
  DashboardData,
  DashboardKpi,
  FleetStatus,
  ActivityEvent,
  PerformanceMetrics
} from '../store/dashboard/dashboard.state';

/**
 * T014: Dashboard API service for HTTP operations.
 * Feature: 022-dashboard-real-data
 */
@Injectable({
  providedIn: 'root'
})
export class DashboardApiService {
  private readonly http = inject(HttpClient);
  private readonly baseUrl = `${environment.apiUrl}/location/v1/admin/dashboard`;

  /**
   * Get all dashboard data in a single API call.
   */
  getDashboardData(performancePeriod: 'week' | 'month' = 'week'): Observable<DashboardData> {
    const params = new HttpParams().set('performancePeriod', performancePeriod);
    return this.http.get<DashboardData>(this.baseUrl, { params });
  }

  /**
   * Get KPIs only.
   */
  getKpis(): Observable<DashboardKpi> {
    return this.http.get<DashboardKpi>(`${this.baseUrl}/kpis`);
  }

  /**
   * Get fleet status breakdown.
   */
  getFleetStatus(): Observable<FleetStatus> {
    return this.http.get<FleetStatus>(`${this.baseUrl}/fleet-status`);
  }

  /**
   * Get recent activity feed.
   */
  getActivity(limit: number = 5): Observable<ActivityEvent[]> {
    const params = new HttpParams().set('limit', limit.toString());
    return this.http.get<ActivityEvent[]>(`${this.baseUrl}/activity`, { params });
  }

  /**
   * Get performance metrics.
   */
  getPerformance(period: 'week' | 'month' = 'week'): Observable<PerformanceMetrics> {
    const params = new HttpParams().set('period', period);
    return this.http.get<PerformanceMetrics>(`${this.baseUrl}/performance`, { params });
  }
}
